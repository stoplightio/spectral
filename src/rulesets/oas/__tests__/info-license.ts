import type { Spectral } from '../../../spectral';
import { createWithRules } from './__helpers__/createWithRules';

describe('info-license', () => {
  let s: Spectral;

  beforeEach(async () => {
    s = await createWithRules(['info-license']);
  });

  test('validate a correct object', async () => {
    const results = await s.run({
      swagger: '2.0',
      paths: {},
      info: {
        contact: { name: 'stoplight.io' },
        license: { name: 'MIT' },
      },
    });
    expect(results.length).toEqual(0);
  });

  test('return errors if info missing license', async () => {
    const results = await s.run({
      swagger: '2.0',
      paths: {},
      info: {
        contact: { name: 'stoplight.io' },
      },
    });
    expect(results).toEqual([
      {
        code: 'info-license',
        message: 'OpenAPI object `info` should contain a `license` object.',
        path: ['info'],
        range: {
          end: {
            character: 28,
            line: 5,
          },
          start: {
            character: 9,
            line: 3,
          },
        },
        severity: 1,
      },
    ]);
  });
});
