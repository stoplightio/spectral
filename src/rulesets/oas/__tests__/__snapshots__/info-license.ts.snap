// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info-license return errors if info missing license 1`] = `
Array [
  Object {
    "code": "info-license",
    "message": "OpenAPI object info \`license\` must be present and non-empty string.",
    "path": Array [
      "info",
      "license",
    ],
    "range": Object {
      "end": Object {
        "character": 28,
        "line": 5,
      },
      "start": Object {
        "character": 9,
        "line": 3,
      },
    },
    "severity": 1,
    "source": undefined,
    "summary": "OpenAPI object info \`license\` must be present and non-empty string.",
  },
]
`;
